### PR TITLE
Fix #2759: Removed profile_java_proto_lite from the model exports

### LINKED ISSUE #2759: Remove :profile_java_proto_lite from the list of exports in the model library
This task is part of the build system migration from Gradle to Bazel. A prerequesite before working on this issue is that you are able to build Oppia using Bazel. Check out [Oppia Bazel guide](https://github.com/oppia/oppia-android/wiki/Oppia-Bazel-Guide) for Bazel terminology and more on what Bazel does.

## 1. Remove the export

In 

`/oppia-android/model/BUILD.bazel`

Remove :profile_java_proto_lite from the list of exports in the model library

```BUILD
android_library(
    name = "model",
    visibility = ["//visibility:public"],
    exports = [
        ... # <- Remove it from this list.
    ],
)
```

## 2. Add public visibility to the java_lite_proto_library

Add 

`    visibility = ["//visibility:public"],`

to the java_lite_proto_library that has the name profile_java_proto_lite.

## 3. Build the app using Bazel and run the tests

Libraries that are indirectly depending on `profile_java_proto_lite` will now fail to build. Try to build the entire app with Bazel and add the deps dependency

`//model:profile_java_proto_lite,`

to all the libraries that are now failing to build. You will have to look around for BUILD.bazel files that have libraries that require this replaced dependency.

These libraries will already have the //model dependency and since we removed the :profile_java_proto_lite from the model library itself, they will need to add the newly available public library directly as a dependency. (Not all libraries with the //model dependency may require the newly available library)

Once the app builds and all tests pass, you're done!

### PR DESCRIPTION
**Pull request by jonathanalvares9009**
## Explanation

This PR will remove profile_java_proto_lite from the model BUILD.bazel file and add the dependency to the dependents in other modules individually. 
Fixes #2759 

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by jonathanalvares9009** [comment]
@BenHenning and @vinitamurthi can you review my PR?

**Comment by jonathanalvares9009** [comment]
@fsharpasharp can you review my PR?

**Review by fsharpasharp — APPROVED** [review]
Looks good!

File: model/BUILD.bazel [hunk 1]
```diff
@@ -89,6 +89,7 @@ proto_library(
 
 java_lite_proto_library(
     name = "profile_java_proto_lite",
+    visibility = ["//visibility:public"],
```
Comment (vinitamurthi) [inline model/BUILD.bazel]: Is this required? As much as possible we would like to avoid setting visibility to public

**Review by vinitamurthi — CHANGES_REQUESTED** [review]

File: model/BUILD.bazel (hunk 1 above)
Comment (jonathanalvares9009) [reply model/BUILD.bazel]: @vinitamurthi it was written in the issue to make it public. But I'll try API.

Comment (jonathanalvares9009) [reply model/BUILD.bazel]: @vinitamurthi changed visibility label to oppia_api_visibilty from public

**Comment by jonathanalvares9009** [comment]
@vinitamurthi can you please tell me why public visibility is avoided? Just so that I don't repeat the mistake again

**Comment by jonathanalvares9009** [comment]
@vinitamurthi Ig changing the visibility label fails the tests coz the app builds after this change. So should I change it back to public or leave it this way?

**Comment by vinitamurthi** [comment]
> @vinitamurthi can you please tell me why public visibility is avoided? Just so that I don't repeat the mistake again

Public basically gives any and every module access to this module. If we restrict visibility, we have more control over who can access this module, and we only deliberately give access when its really required.
@fsharpasharp since you added this issue, I wanted to check if visibility:public is actually required for this scenario or can we restrict the visibility to a smaller group of modules? @BenHenning let me know your thoughts as well

**Comment by jonathanalvares9009** [comment]
@vinitamurthi, @BenHenning, and @fsharpasharp please let me know what the visibility should be.

**Comment by fsharpasharp** [comment]
> > @vinitamurthi can you please tell me why public visibility is avoided? Just so that I don't repeat the mistake again
> 
> Public basically gives any and every module access to this module. If we restrict visibility, we have more control over who can access this module, and we only deliberately give access when its really required.
> @fsharpasharp since you added this issue, I wanted to check if visibility:public is actually required for this scenario or can we restrict the visibility to a smaller group of modules? @BenHenning let me know your thoughts as well

I second your first paragraph here that visibilities should be restricted as much as possible. @jonathanalvares9009, some of these issues were filed when we didn't have the defined specific visibilities. I haven't updated these issues to be consistent with this new practice. If possible, we would prefer the more informative and restrictive visibility for each library. //visibility:public is not a good visibility setting since it is too broad and does not signal the intended usage, so I'm all for using an alternative here.

For the failing tests, I think https://github.com/oppia/oppia-android/issues/3018 could be a cause and you could revisit this as soon as that gets fixed.

**Comment by jonathanalvares9009** [comment]
So @fsharpasharp my PR is correct right coz I made the visibly oppia_api_visibilty.

**Review by BenHenning — APPROVED** [review]
Thanks @jonathanalvares9009! This LGTM.

**Comment by BenHenning** [comment]
The visibility adjustment SGTM, and API visibility seems like the correct long-term visibility contract for proto models.

**Review by vinitamurthi — APPROVED** [review]

**Comment by vinitamurthi** [comment]
LGTM! Note that some test is failing @jonathanalvares9009 . Once you fix that we can merge this

**Comment by jonathanalvares9009** [comment]
@vinitamurthi can you help me with the bazel tests I ran it multiple times on my local machine it runs successfully.

**Comment by vinitamurthi** [comment]
Can you try merging the latest version of develop into your branch and then pushing it here? The run seems to be 2 days old

**Comment by jonathanalvares9009** [comment]
@vinitamurthi same thing is happening again

**Comment by vinitamurthi** [comment]
Did you try running bazel build on your local machine? The build seems to be failing and I am not sure why it would fail on CI if its passing for you locally. @BenHenning any thoughts?

**Comment by jonathanalvares9009** [comment]
Screenshot 2021-03-31 at 9 58 39 AM

@vinitamurthi and @BenHenning it's running locally as you can see from the screenshot. I don' know why it's failing CI tests

**Comment by fsharpasharp** [comment]
> Screenshot 2021-03-31 at 9 58 39 AM
> 
> @vinitamurthi and @BenHenning it's running locally as you can see from the screenshot. I don' know why it's failing CI tests

The command you are running just builds the app, it doesn't run the tests. See https://github.com/oppia/oppia-android/wiki/Oppia-Bazel-Setup-Instructions for some more context on running the tests.

Edit: I think the failing tests are still part of an issue that is tracked here: https://github.com/oppia/oppia-android/issues/3018.

**Comment by jonathanalvares9009** [comment]
@fsharpasharp should I try running bazel test //...

**Comment by BenHenning** [comment]
@jonathanalvares9009 I wouldn't recommend that--it takes a pretty large amount of memory currently to run the whole suite. I suggest instead to merge in the latest develop and try running: ``bazel test //data/...`` since the data module was the only one affected by this change. Make sure to merge in develop again, though, since #3025 has fixed all Bazel tests unrelated to this PR.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Marking 'requested changes' until we verify that //data/... is passing. @jonathanalvares9009 please provide a screenshot of those tests passing locally after you merge in the latest develop changes.

Please make sure to also push those changes to GitHub so that the CI checks can start passing, too.

**Comment by jonathanalvares9009** [comment]
Screenshot 2021-04-01 at 11 04 37 AM

**Comment by BenHenning** [comment]
Ah, darn. It seems you may have run into JCenter flaking out again, but it's hard to tell from the error.

**Review by BenHenning — APPROVED** [review]
//data tests are passing; the single failure is a known flake being fixed by #2916.

Thanks @jonathanalvares9009!

**Pull request merged**